### ISSUE TITLE
Password encryption for log in page

### ISSUE THREAD

**Issue by sureshab** [kimchi-project/wok#134]
Password is sent as plain text from log in page to backend, below are few security issues which I can think of-
1) When the request lands on controller or model, any admin user can have debug statements and read user credentials.
2) java script is not completely minified, so there can be possibility to modify javascript and hack user credentials.
3) Plugins like firebug can read the user name and password, the same when compared with gmail log in page, it bypasses firebug.

**Comment by danielhb** [comment]
About (2) and (3), when you access WoK in <ip>:8000 , you are redirected to https:<ip>:8001 . From that moment on, the communication is secure using SSL. So the password does not get intercepted in its way to the server as plain text due to SSL.

About Firebug, we can't do much to prevent it. It's all client side code and it's modifiable. If the user modifies the input type="password" to input type="text" then the password would be visible in plain sight. And this is easily done by using Firebug or Chrome Dev Tools. Both methods can also read the information behind input type="password" too. 

In short, we can guarantee that the password will not be intercepted middle way to the server and that the server does not log the plain text password (your first point). Other than that we can't do much else because, again, when the page is loaded in the browser, we can't control its contents.

**Comment by chandrureddy** [comment]
@danielhb : 
I agree.  But on the other hand there are some crypto modules helps in hashing the password etc ... If I remember correctly node.js has such an option. 

We need to think in those lines where any of the existing options help us in preventing such impacts. 

One good idea is minifying the javascripts if it helps some extent.

**Comment by danielhb** [comment]
I am all for anything that minimizes the impact. We just need to have in mind that if the user wants to tamper with the HTML5 or JS files in its side (client side), there's nothing we can do to prevent it

**Comment by chandrureddy** [comment]
What will happen when attribute "https_only = false " enabled in the wok.conf  ?
Will this become a security breach if user start using http instead https ?

**Comment by alinefm** [comment]
@chandrureddy when setting 'https_only = false' means the HTTP port (8000 by default) will still be accessible BUT all HTTP requests are redirected to HTTPS port (8001 by default).
You can check that while trying to accessing wok on 8000, it will be automatically redirected to 8001 (HTTPS).

**Comment by sureshab** [comment]
@alinefm 
i have a question, not related to security though,,, what is the significance of https_only flag?
the only difference i noticed is, when flag is set to false, connect to http://<ip>:8000 will redirect to https and 80001 port, but end of the day no requests are processed in http://<ip>:8000. 
So why wasn't it restricted to just https always? please correct me if i am missing something.

**Comment by alinefm** [comment]
Hi @sureshab 

The https_only option was added 'recently' as requested by @chandrureddy to do not allow HTTP connection.
About make it the default behavior, we need to discuss that with the users and community.

I will close this issue as it is not a bug.
